### PR TITLE
add dataset & scopes to token

### PR DESCRIPTION
**Pull request by ranbreuer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,3 +87,45 @@ The following decoded JSON web token
   "nbf": 1360043456
 }
 ```
+
+## Adding Permission Scopes to Embed Tokens
+When using Embed tokens, one might want to restrict usage of the resources he gives access to. For this reason, you can generate a token with scoped permissions.
```
Comment (yuryb2) [inline README.md]: This is public documentation - do we need to review it with technical writers?

Comment (yuryb2) [inline README.md]: This is public documentation - do we need to review it with technical writers?

Comment (yuryb2) [inline README.md]: This is public documentation - do we need to review it with technical writers?

Comment (ranbreuer) [inline README.md]: we do, will do this as part of merge to master

File: lib/core/powerBIToken.ts [hunk 2]
```diff
@@ -14,18 +14,28 @@ export class PowerBIToken {
...
         token.addClaim('wid', workspaceId);
-        token.addClaim('rid', reportId);
 
+        if (reportId) {
+            token.addClaim('rid', reportId);
+        }
+
+        if (datasetId) {
+            token.addClaim('did', datasetId);
+        }
```
Comment (yuryb2) [inline lib/core/powerBIToken.ts]: What if you have both rid and did?

File: lib/core/powerBIToken.ts [hunk 3]
```diff
@@ -34,6 +44,14 @@ export class PowerBIToken {
             }
         }
 
+        if (scopes) {
```
Comment (yuryb2) [inline lib/core/powerBIToken.ts]: Is it valid not passing any scopes?

Comment (ranbreuer) [inline lib/core/powerBIToken.ts]: reverting to defaults

**Pull request merged**